src/encode.cpp

```diff
@@ -117,6 +117,63 @@ static mlir::Type getTensorElemTy(mlir::Value v) {
   return v.getType().dyn_cast<mlir::TensorType>().getElementType();
 }
 
+static pair<Tensor, Tensor> 
+getBroadcastTensor(State &st, mlir::Value arg0, mlir::Value arg1) {
+  // reference: https://numpy.org/doc/stable/user/basics.broadcasting.html
+  auto ty0 = arg0.getType().cast<mlir::RankedTensorType>();
+  auto ty1 = arg1.getType().cast<mlir::RankedTensorType>();
+  auto resRank = max(ty0.getRank(), ty1.getRank());
+  auto inVars0 = Index::boundIndexVars(resRank);
+  auto inVars1 = Index::boundIndexVars(resRank);
+
+  vector<Expr> outVars0, outVars1, resDims;
+  for(int i = 0; i < resRank; i++) {
+    auto idx0 = ty0.getRank() - 1 - i;
+    auto idx1 = ty1.getRank() - 1 - i;
+
+    if(idx0 >= 0 && idx1 >= 0) {
```

Comment:
idx0 and idx1 are unsigned integers..?

Reply:
getRank returns int64_t, so I think it would be int64_t, but I'll cast it with int64_t